### DISCUSSION TITLE
Uniscribe and/or Core Text support?

### DISCUSSION THREAD

**Discussion by bdenckla** [fonttools/fontbakery#3235] [Ideas]
Does fontbakery have any plans to add Uniscribe and/or Core Text support? Or will it stick with only HarfBuzz "forever"?

Do Uniscribe and Core Text even have the "hooks" that fontbakery would need?

As documented on its [Uniscribe integration](https://harfbuzz.github.io/integration-uniscribe.html) page, HarfBuzz tries to behave as similarly to Uniscribe as possible.

So perhaps adding Uniscribe support to fontbakery would not offer as much "bang for buck" as adding Core Text support.

But I find that HarfBuzz behaves quite differently from Uniscribe in one way very important to me: HarfBuzz performs Unicode normalization (or something quite close to it) on its input strings, whereas Uniscribe does not.

So for me, Uniscribe support in fontbakery would be very helpful. For more background on how HarfBuzz's normalization hampers testing of fonts for Biblical Hebrew, see [this HarfBuzz GitHub discussion I started](https://github.com/harfbuzz/harfbuzz/discussions/2942).

**Comment by bdenckla** [comment]
Update: [a pull request](https://github.com/googlefonts/fontbakery/pull/3236) by @simoncozens implementing subshaper selection has been merged.

**Reply by simoncozens** [reply]
Right. Please note that fontbakery is a very general and wide ranging font QA tool, and the shaper check is just one out of hundreds of checks that it performs. I think allowing subshaper selection in the JSON files is the best we can do given that the other shapers are not open source and are platform specific. People wishing to test shaping behaviour on different platforms will need to run fontbakery on each platform, using the subshaper selection. DirectWrite support in the Python uharfbuzz library is an open issue, but OS X and Linux can both be checked by running on OSX with subshaper selection.
